lib/models/issue.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+class Issue < ActiveRecord::Base
+  belongs_to :route
+  belongs_to :stop
+
+  serialize :alternatives, Array
+
+  validates :headsign, :sdt, presence: true
+  validates :issue_type, inclusion: { in: %w[missing incorrect] }
+
+  scope :visible, -> { where visible: true }
+
+  def format_sdt(sdt)
+    Time.at(sdt * 60).utc.strftime '%l:%M %P'
+  end
+
+  def new?
+    github_number.blank?
+  end
+
+  def text
+    opener = 'This error has recurred.' unless open? && visible?
+    body = <<~TEXT
+      #{Issue.timestamp}: Route #{route.number} with headsign #{headsign} is #{issue_type}.
+      Expected to be departing from #{stop.name} at #{format_sdt(sdt)}.
+    TEXT
+    if alternatives.present?
+      closer = case issue_type
+               when 'incorrect'
+                 "Saw SDT: #{alternatives.map { |t| format_sdt(t) }.join(', ')}"
+               when 'missing'
+                 "Found alternatives: #{alternatives.join(', ')}"
+               end
+    end
+    [opener, body, closer].compact.join "\n"
+  end
+
+  def title
+    "#{stop.name}: #{issue_type} #{route.number} #{headsign} SDT"
```

Comment:
@sherson you had mentioned a preference about issue title before. Does this seem like an acceptable format?

Reply:
@umts/umts-tpa what format would be most convenient for you? If it were me, I think I'd want something like:

`#{route.number} / #{headsign} / #{stop.name} / SDT / #{issue_type}`